File: src/main/java/org/activecraft/activecraftcore/guicreator/GuiCancelEvent.java
package org.activecraft.activecraftcore.guicreator;

import org.activecraft.activecraftcore.events.ActiveCraftEvent;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import org.activecraft.activecraftcore.events.ActiveCraftEvent;
import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;

@Data
@EqualsAndHashCode(callSuper = false)
public class GuiCancelEvent extends ActiveCraftEvent {

    private boolean cancelled;
    private Gui gui;
    private @NonNull Inventory clickedInventory;
    private @NonNull String identifier;
    private @NonNull Player player;

    public Gui getGui() {
        return gui != null ? gui : (gui = Gui.ofInventory(clickedInventory));
    }
}

File: src/main/java/org/activecraft/activecraftcore/guicreator/GuiCancelEvent.kt
package org.activecraft.activecraftcore.guicreator

import lombok.Data
import lombok.EqualsAndHashCode
import org.activecraft.activecraftcore.events.ActiveCraftEvent
import org.activecraft.activecraftcore.events.CancellableActiveCraftEvent
import org.activecraft.activecraftcore.guicreator.Gui.Companion.ofInventory
import org.bukkit.entity.Player
import org.bukkit.inventory.Inventory

@Data
@EqualsAndHashCode(callSuper = false)
class GuiCancelEvent(val clickedInventory: Inventory, val identifier: String, val player: Player) : CancellableActiveCraftEvent() {
    var gui: Gui? = null
        get() = if (field != null) field else ofInventory(clickedInventory).also { field = it }
        private set
}